### DISCUSSION TITLE
discuss about redis pool optimize

### DISCUSSION THREAD

**Discussion by andrewshan** [polarismesh/polaris#416] [General]
Now that, some issues and pr about redis pool, here is 
#415 support redis sentinel
#398 support TLS configuration
#404 configuration about some connections

there are three pull requests about those issue, some of them has different implementations, so we need to discuss about the best

#408 
#403 
#405

**Comment by magederek** [comment]
The code style of `WithOptions()` is extendable more easily. The refactor is more suitable to be applied fully backward compatible without any options changed. The mode of func options can also be considered to apply to other plugins's config, e.g. auth.
